cli/src/commands/branch.rs

```diff
@@ -96,18 +96,24 @@ pub struct BranchDeleteArgs {
 /// https://github.com/martinvonz/jj/blob/main/docs/branches.md.
 #[derive(clap::Args, Clone, Debug)]
 pub struct BranchListArgs {
+    /// DEPRECATED: use `--all-remotes` instead.
     /// Show all tracking and non-tracking remote branches including the ones
     /// whose targets are synchronized with the local branches.
-    #[arg(long, short, conflicts_with_all = ["names", "tracked"])]
+    #[arg(long, conflicts_with_all = ["names", "tracked"], hide = true)]
     all: bool,
 
+    /// Show all tracking and non-tracking remote branches including the ones
+    /// whose targets are synchronized with the local branches.
+    #[arg(long, short, conflicts_with_all = ["names", "tracked"])]
+    all_remotes: bool,
```

Comment:
I think a hidden alias without deprecation (i.e. `alias = "all"`) is okay. What do you think?

Reply:
Works for me. There's no direct need to deprecate the old style. Will update the PR

Reply:
I have updated the PR to go with alias instead of deprecation